File: solutions/219/main.py
from typing import List 

class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        my_set = []
        for i in range(len(nums)):
            if nums[i] in my_set:
                return True
            my_set.append(nums[i])
            if i >= k:
                my_set.pop(0)
        return False

if __name__ == "__main__":
    nums = [1,2,3,1]
    k = 3 
    s = Solution()
    assert s.containsNearbyDuplicate(nums, k)

File: solutions/219/main.rs
use std::collections::HashSet; 

struct Solution {}

impl Solution {
    pub fn contains_nearby_duplicate(nums: Vec<i32>, k: i32) -> bool {
        let mut my_set = HashSet::new(); 
        for i in 0..nums.len() {
            if my_set.contains(&nums[i]) {
                return true; 
            }
            my_set.insert(nums[i]);
            if i >= k as usize {
                my_set.remove(&nums[i-k as usize]);
            }
        }
        return false; 
    }
}

fn main() {
    assert!(Solution::contains_nearby_duplicate(vec![1,2,3,1], 3))
}
